R/build.r

```diff
@@ -278,15 +291,22 @@ build_site <- function(pkg = ".",
                        lazy = FALSE,
                        override = list(),
                        preview = NA,
-                       new_process = TRUE,
-                       devel = TRUE,
+                       devel = FALSE,
+                       new_process = !devel,
                        document = "DEPRECATED") {
+  pkg <- as_pkgdown(pkg)
 
   if (!missing(document)) {
     warning("`document` is deprecated. Please use `devel` instead.", call. = FALSE)
     devel <- document
   }
 
+  if (!devel) {
+    withr::local_temp_libpaths()
+    rule("Installing temporary version of package")
+    utils::install.packages(pkg$src_path, repo = NULL, type = "source", quiet = TRUE)
```

Comment:
Are you sure that the temporary libpath will propagate into the sub process? If it doesn't, you probably wont notice it because it silently picks up on another installation. 

Maybe `quiet = TRUE` is  dangerous because the user see what's wrong if the installation fails (and again it will silently pick up on another installation.

Reply:
Points noted about `install.packages()`, but I'm not sure how to do better without introducing a bunch of dependencies. This fix resolves an important problem (your articles building from an outdate package), so I think it's an improvement over the current state, and worth merging (and we come back to in the future if we find a better approach).

Reply:
Yes it makes sense. It's fine by me if you don't mind introducing another parameter `install` that we can use in cases where we want to manually verbosely install first, and then generate the site after that.